forge-core/src/main/java/forge/card/CardRules.java

```diff
@@ -275,18 +275,53 @@ public boolean canBeCommander() {
         return false;
     }
 
+    public boolean canBePartnerCommanders(CardRules b) {
+        if (!(canBePartnerCommander() && b.canBePartnerCommander())) {
+            return false;
+        }
+        boolean legal = false;
+        if (hasKeyword("Partner") && b.hasKeyword("Partner")) {
+            legal = true; // normal partner commander
+        }
+        if (getName().equals(b.getPartnerWith()) && b.getName().equals(getPartnerWith())) {
+            legal = true; // paired partner commander
+        }
+        if (hasKeyword("Friends forever") && b.hasKeyword("Friends forever")) {
+            legal = true; // Stranger Things Secret Lair gimmick partner commander
+        }
+        if (hasKeyword("Choose a Background") && b.canBeBackground()
+                || b.hasKeyword("Choose a Background") && canBeBackground()) {
+            legal = true; // commander with background
+        }
+        if (isDoctor() && b.hasKeyword("Doctor's companion")
+                || hasKeyword("Doctor's companion") && b.isDoctor()) {
```

Comment:
IIRC @tool4ever updated to use `haskeyword(Keyword.key)`, maybe just use `hasKeyword(Keyword.DOCTORS_COMPANION)` and maybe update others too..

Reply:
Doesn't work
Only Card has `hasKeyword(KeywordEnum)`

CardRules only has `hasKeyword(String)`

Also the KeywordEnum only exist in Forge-Game, not Forge-Core

Reply:
@tehdiplomat we need to think about it if we really still need Forge-Core or if we should merge this with Forge-Game